src/containers/router.js

```diff
@@ -34,6 +34,7 @@ export default class Root extends Component {
 
   componentWillMount () {
     let currentPath = store.getState().routing.locationBeforeTransitions.pathname;
+    window.history.replaceState('', '', currentPath);
```

Comment:
Yes. But now search param is not working, so... don't merge it. I didn't know it was done/working, so I need to take a look at it. The *double* routing is a bit problematic.